duckduckgo/Android
# Search with Favorites widget

Task/Issue URL: https://app.asana.com/0/488551667048375/1200651343654716/f
Tech Design URL: 
CC: 

**Description**:

This PR introduces a new widget for Android. The new widget will show user's favorites and the search bar.
The widget empty case includes a CTA to take the user through a small onboarding about how to add favorites.

We've also updated the UI for the existing widgets. 


**Steps to test this PR**:

- Test 1: onboarding

1. Fresh install
2. Navigate through the onboarding until you see the widget CTA (visit a site, open a new tab, visit another site, open another tab)
1. Click on "Add Wiget" and then "Add automatically"
2. press home and go to your homescreen
3. Ensure widget was added
4. Widget should a CTA saying "No Favorites added yet" - "Add Favorite"
5. Click on "Add Favorite"
6. Ensure pixel "m_sfew_l" sent
7. Ensure Dax CTA appears "Visit your favorite sites in a flash! ..."
8. Ensure pixel "m_fo_s" sent
9. Visit a Site
10. Ensure overflow menu is highlighted
11. Open the menu
12. Ensure "Add Favorite" is highlighted
13. Click on that option
14. Ensure pixel "m_nav_af_p" sent with "fmi=true"
15. Visit another site
16. Add the new site as favorite
14. Ensure pixel "m_nav_af_p" sent with "fmi=false"
15. Press home
16. Ensure new favorites sites are shown in the widget
14. Now click on the widget "search bar"
15. Ensure you go to SystemSearch screen
16. Ensure Pixel "m_sfbw_l" sent

- Test 2: interact with favorites

1. click on any favorite from the widget
2. Ensure pixel "m_sfiw_l" sent
3. Focus the URL bar
4. Quick access grid should appear, visit a favorite from there
5. Ensure pixel "m_fav_o" sent
6. Now open a new tab 
7. Ensure pixel "m_fav_ht" sent
8. Go to bookmarks screen
9. Click on a favorite
10. ensure pixel "m_fav_b" sent
11. Remove one of your favorites
12. Go to your homescreen
13. Ensure the widget doesn't show the favorite you removed

- Test 3: Add widget manually

1. Add the favorites widget manually on your home screen (no need to remove the previous one)
2. When the widget is placed on your homescreen, the configuration screen should appear
3. Ensure pixel "m_sfw_cs" is sent
4. Select "system default" theme and "add widget"
5. Ensure pixel "m_sfw_sd" sent
6. Change the theme of your device
7. Widget should also change to honor system theme

- Test 4: Resize the widget
1. You should be able to resize the widget (width or height)
2. Reduce the widget width to only show 2 columns
3. Ensure search bar hint is not visible
4. If widget has not favorites, empty CTA will not appear when widget has 2 columns


---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
